dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -864,6 +870,36 @@ public List<String> getAllHeaderNames() {
         return vals;
     }
 
+    /**
+     * This method is the same as getAllHeaderNames, but returns a Set instead.
+     * This was done in order to avoid calling contains on every header name.
+     *
+     * @see com.ibm.wsspi.genericbnf.HeaderStorage#getAllHeaderNamesSet()
+     */
+    @Override
+    public Set<String> getAllHeaderNamesSet() {
```

Comment:
Would getAllHeaderNames be faster if we collected the names with getAllHeaderNamesSet and used the resulting set to create the ArrayList it needs?